zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -318,6 +318,253 @@ def test_security_checklist(self):
+
+        set_default = {charm_key: vpair[0]}
+        set_alternate = {charm_key: vpair[1]}
+        app_name = service
+
+        conf_file = '/etc/neutron/plugins/ml2/openvswitch_agent.ini'
+
+        expected = {
+            section: {
+                config_file_key: [vpair[1]],
+            },
+        }
+
+        with self.config_change(set_default,
+                                set_alternate,
+                                application_name=app_name):
+            zaza.model.block_until_oslo_config_entries_match(
+                self.application_name,
+                conf_file,
+                expected,
+            )
+        logging.debug('OK')
+
+    def test_l2pop_propagation(self):
+        """Verify that l2pop setting propagates to neutron-ovs."""
+        # Needs love - not idempotent
+        self._check_ml2_setting_propagation('neutron-api',
+                                            'l2-population',
+                                            'l2_population',
+                                            ['False', 'True'],
+                                            'agent')
+
+    def test_nettype_propagation(self):
+        """Verify that nettype setting propagates to neutron-ovs."""
+        # Needs love - not idempotent
+        self._check_ml2_setting_propagation('neutron-api',
+                                            'overlay-network-type',
+                                            'tunnel_types',
+                                            ['vxlan', 'gre'],
+                                            'agent')
+
+    def test_secgroup_propagation_local_override(self):
+        """Verify disable-security-groups overrides what neutron-api says."""
+        if self.current_os_release >= self.trusty_mitaka:
+            conf_file = "/etc/neutron/plugins/ml2/openvswitch_agent.ini"
+        else:
+            conf_file = "/etc/neutron/plugins/ml2/ml2_conf.ini"
+
+        zaza.model.set_application_config(
+            'neutron-api',
+            {'neutron-security-groups': 'True'})
+        zaza.model.set_application_config(
+            'neutron-openvswitch',
+            {'disable-security-groups': 'True'})
+
+        zaza.model.wait_for_application_states()
+
+        expected = {
+            'securitygroup': {
+                'enable_security_group': ['False'],
+            },
+        }
+
+        zaza.model.block_until_oslo_config_entries_match(
+            self.application_name,
+            conf_file,
+            expected,
+        )
+
+        logging.info('Restoring to default configuration...')
+        zaza.model.set_application_config(
+            'neutron-openvswitch',
+            {'disable-security-groups': 'False'})
+        zaza.model.set_application_config(
+            'neutron-api',
+            {'neutron-security-groups': 'False'})
+
+        zaza.model.wait_for_application_states()
+
+    def test_z_restart_on_config_change(self):
```

Comment:
This should be numbered. I know it is not in the original but it one of the tests.